pkg/resources/server_test.go

```diff
@@ -0,0 +1,327 @@
+			err := rs.Start()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("Stop", func() {
+		It("stop server with correct parameters", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("DeleteServer").Return()
+			rsc.On("GetServer").Return(grpcServer)
+			rsc.On("Stop").Return()
+
+			stopWatcher := make(chan bool)
+			rs := resourceServer{
+				rsConnector: rsc,
+				stopWatcher: stopWatcher,
+				stop:        make(chan interface{}),
+			}
+			// Dummy listener to stopWatcher to not block the test and fail
+			go func() {
+				<-stopWatcher
+			}()
+
+			err := rs.Stop()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("stop non existing server", func() {
+			rs := resourceServer{}
+			err := rs.Stop()
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+	Context("Restart", func() {
+		It("Restart server with correct parameters", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("GetServer", mock.Anything).Return(grpcServer)
+			rsc.On("CreateServer").Return()
+			rsc.On("DeleteServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Serve", mock.Anything).Return()
+			rsc.On("Dial", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Stop").Return()
+			rsc.On("Close", mock.Anything).Return()
+
+			rs := resourceServer{
+				watchMode:   true,
+				rsConnector: rsc,
+				stop:        make(chan interface{}),
+			}
+
+			err := rs.Restart()
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+	Context("Watch", func() {
+		var DepSock string
+		var fs utils.FakeFilesystem
+		var cleanTemp func()
+		BeforeSuite(func() {
+			DepSock = deprecatedSockDir
+		})
+		AfterSuite(func() {
+			deprecatedSockDir = DepSock
+		})
+		BeforeEach(func() {
+			fs = utils.FakeFilesystem{}
+			cleanTemp = fs.Use()
+			deprecatedSockDir = fs.RootDir
+		})
+		AfterEach(func() {
+			cleanTemp()
+		})
+		It("Watch socket and send notification of create successfully", func() {
+			grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("GetServer", mock.Anything).Return(grpcServer)
+			rsc.On("CreateServer").Return()
+			rsc.On("DeleteServer").Return()
+			rsc.On("Listen", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Serve", mock.Anything).Return()
+			rsc.On("Dial", mock.Anything, mock.Anything).Return(nil, nil)
+			rsc.On("Stop").Return()
+			rsc.On("Close", mock.Anything).Return()
+
+			rs := resourceServer{
+				watchMode:   true,
+				rsConnector: rsc,
+				socketPath:  "fake",
+				stop:        make(chan interface{}),
+			}
+			go func() {
+				// Wait for watch to create the socket watcher
+				time.Sleep(5 * time.Millisecond)
+				event := fsnotify.Event{Name: "fake", Op: fsnotify.Create}
+				rs.socketWatcher.Events <- event
+			}()
+			err := rs.Watch()
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
imo need to ensure Restart() was called here as well

Reply:
add AssertExpectations to make sure all mocked functions all called